src/H5Dmpio.c

```diff
@@ -5042,29 +5621,32 @@ H5D__mpio_get_chunk_redistribute_info_types(MPI_Datatype *contig_type, bool *con
     if (H5F_mpi_get_file_block_type(false, &chunk_block_type, &chunk_block_type_derived) < 0)
         HGOTO_ERROR(H5E_DATASET, H5E_CANTGET, FAIL, "can't create derived type for chunk file description");
 
-    field_count = 5;
+    field_count = 6;
     assert(field_count == (sizeof(types) / sizeof(MPI_Datatype)));
 
     /*
      * Create structure type to pack chunk H5F_block_t structure
-     * next to chunk_idx, orig_owner, new_owner and num_writers
-     * fields
+     * next to chunk_idx, dset_oloc_addr, orig_owner, new_owner
+     * and num_writers fields
      */
     block_lengths[0] = 1;
     block_lengths[1] = 1;
     block_lengths[2] = 1;
     block_lengths[3] = 1;
     block_lengths[4] = 1;
+    block_lengths[5] = 1;
     displacements[0] = offsetof(H5D_chunk_redistribute_info_t, chunk_block);
     displacements[1] = offsetof(H5D_chunk_redistribute_info_t, chunk_idx);
-    displacements[2] = offsetof(H5D_chunk_redistribute_info_t, orig_owner);
-    displacements[3] = offsetof(H5D_chunk_redistribute_info_t, new_owner);
-    displacements[4] = offsetof(H5D_chunk_redistribute_info_t, num_writers);
+    displacements[2] = offsetof(H5D_chunk_redistribute_info_t, dset_oloc_addr);
```

Comment:
When sending information about chunks between MPI ranks, the parallel compression code now also needs to send the dataset object header address for the chunk as well so that we can distinguish between two chunks that have the same chunk index value.